tmc-langs-qmake/src/main/java/fi/helsinki/cs/tmc/langs/qmake/QTestResultParser.java

```diff
@@ -0,0 +1,204 @@
+     * Points are mapped to test cases with Message node type 'qinfo'. These
+     * messages contain: prefix "TMC:" test case name: "test_function_name"
+     * points separated by period: ".1"
+     *
+     * With failing testcase assertions, there will be an Incident node with
+     * type "fail" and Description node with the failed assertion message.
+     *
+     * When the testcase assertion(s) has passed, there will be an Incident node
+     * with type "pass" and no Description node.
+     * </p>
+     * <TestFunction name="test_function_name">
+     * <Message type="qinfo" file="" line="0">
+     * <Description><![CDATA[TMC:test_function_name.1]]></Description>
+     * </Message>
+     * <Incident type="fail" file="test_source.cpp" line="420">
+     * <Description>
+     * <![CDATA['!strcmp(hello_msg(), "Helo, world!" )' returned FALSE. ()]]>
+     * </Description>
+     * </Incident>
+     * <Duration msecs="0.135260"/>
+     * </TestFunction>
+     *
+     */
+    private List<TestResult> createQtTestResults(NodeList nodeList) {
+        List<TestResult> cases = new ArrayList<>();
+
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            Element testcase = (Element) nodeList.item(i);
+            List<String> points = parsePoints(testcase);
+
+            if (points.isEmpty()) {
+                // No points == not a TMC testcase
+                continue;
+            }
+
+            Element incident = (Element) testcase.getElementsByTagName("Incident").item(0);
+
+            String id = testcase.getAttribute("name");
+            boolean passed = incident.getAttribute("type").equals("pass");
+            String msg = "";
+
+            // Get the assertion error if testcase failed
+            if (!passed) {
+                Element desc = (Element) incident.getElementsByTagName("Description").item(0);
+                msg = desc.getTextContent();
+            }
+
+            ImmutableList<String> trace = ImmutableList.of();
+            cases.add(new TestResult(id, passed, ImmutableList.copyOf(points), msg, trace));
+        }
+
+        return cases;
+    }
+
+    /**
+     * <p>
+     * Parse potential points from testcase.
+     * </p>
+     */
+    private List<String> parsePoints(Element testcase) {
+        List<String> points = new ArrayList<>();
+        NodeList messages = testcase.getElementsByTagName("Message");
+        for (int i = 0; i < messages.getLength(); i++) {
+            // Convert node to element, messages.item(i) returns a node
+            Element message = (Element) messages.item(i);
+            Element desc = (Element) message.getElementsByTagName("Description").item(0);
+            String text = desc.getTextContent();
+            if (text.matches("^(TMC:.*)")) {
+                String[] split = text.split("\\.");
+                String result = split[1];
+                for (int j = 2; j < split.length; j++) {
+                    result += "." + split[j];
+                }
```

Comment:
This is basically skipping the name of the test function and using the rest as the point which could contain a ".", not replacing anything.

I also agree on the 1.8 part.

Reply:
We have already ditched Java 1.7 on master.

Reply:
Sorry for misunderstanding the code the first time, I blame lack of sleep :) Looking at this in more detail, am I correct in my reading that this can be effectively simplified to `String result = text.substring(text.indexOf(".") +1);`?